src/Simulation/Simulators.Tests/TypeExtensionsTest.cs

```diff
@@ -82,12 +104,27 @@ public void ArrayTypes()
             Assert.Equal("[1, 2, 3]", new[] { 1, 2, 3 }.GetNonQubitArgumentsAsString());
             Assert.Equal("[\"foo\", \"bar\"]", new[] { "foo", "bar" }.GetNonQubitArgumentsAsString());
 
-            var opArr = new ICallable[] {
-                new QuantumSimulator().Get<Intrinsic.H>(),
-                new QuantumSimulator().Get<Intrinsic.CNOT>(),
-                new QuantumSimulator().Get<Intrinsic.Ry>(),
+            var simulators = new CommonNativeSimulator[] { 
+                new QuantumSimulator(),
+                new SparseSimulator2(),
+                new QuantumSimulator(),
+                new SparseSimulator2(),
+                new QuantumSimulator(),
+                new SparseSimulator2()
             };
-            Assert.Equal("[H, CNOT, Ry]", opArr.GetNonQubitArgumentsAsString());
+
+            for(uint i = 0; i < 2; ++i)
```

Comment:
This comes across as even more awkward due to instantiating three times. If parameterization isn't feasible, it may be worth considering using actions as an alternative:

```csharp
// NB: pseudocode only
var simulatorFactories = new Func<CommonNativeSimulator>[]
{
    () => new QuantumSimulator(),
    () => new SparseSimulator2()
};

foreach (var factory in simulatorFactories)
{
     varOpArr = new ICallable[]
     {
         factory().Get<Intrinsic.H>(),
         // ...
     };
     Assert.Equal("[H, CNOT, Ry]", opArr.GetNonQubitArgumentsAsString());
}
```

Reply:
Changing to
```c#
Func<CommonNativeSimulator>[] simulatorFactories = new Func<CommonNativeSimulator>[]
{
    () => new QuantumSimulator(),
    () => new SparseSimulator2()
};

// . . .
foreach (var factory in simulatorFactories)
{
    var sim1 = factory();
    var sim2 = factory();
    var sim3 = factory();
    try
    {
        var opArr = new ICallable[] {
            sim1.Get<Intrinsic.H>(),
            sim2.Get<Intrinsic.CNOT>(),
            sim3.Get<Intrinsic.Ry>(),
        };

        Assert.Equal("[H, CNOT, Ry]", opArr.GetNonQubitArgumentsAsString());
    }
    finally
    {
        sim1.Dispose();
        sim2.Dispose();
        sim3.Dispose();
    }
}
```